becquerel/core/spectrum.py

```diff
@@ -929,51 +929,62 @@ def has_uniform_bins(self, use_kev=True):
             raw and kev versions of the bin_edges
 
         Raises:
-          SpectrumError: if use_kev=True but Spectrum is not calibrated
+          UncalibratedError: if use_kev=True but Spectrum is not calibrated
 
         Returns:
-          True/False if all binwidths are equal to within np.finfo(float).eps
+          True/False if all binwidths are equal to within relative tolerance
+            rtol (default 100 * np.finfo(float).eps)
         """
 
+        if rtol is None:
+            rtol = 100 * EPS
+        if rtol < EPS:
+            raise ValueError('Relative tolerance rtol cannot be < system EPS')
+
+        if use_kev is None:
+            use_kev = self.is_calibrated
+
         if use_kev and not self.is_calibrated:
-            raise SpectrumError('Cannot access energy bins with an ' +
-                                'uncalibrated Spectrum.')
+            raise UncalibratedError('Cannot access energy bins with an ' +
+                                    'uncalibrated Spectrum.')
 
         bin_widths = self.bin_widths_kev if use_kev else self.bin_widths_raw
         iterator = iter(bin_widths)
         x0 = next(iterator, None)
         for x in iterator:
-            if abs(x/x0 - 1.0) > EPS:
+            if abs(x/x0 - 1.0) > rtol:
```

Comment:
why not `np.allclose(x, x[0], rtol=rtol)`?

Reply:
I found this nice solution [here](https://github.com/0cjs/py-allthesame#8-writing-the-code-the-function), but if you prefer the `np` solution for readability I can change it. The current version has the advantage of terminating early but I can't imagine that mattering much in an absolute sense.

Reply:
Quick tests indicate that the `np` solution is substantially faster for all but the shortest of actually-uniform arrays, starting around 70 elements, while the `iter` solution is substantially faster for most non-uniform arrays such as log bins. Either way, everything is < 100 us.

Reply:
added a note in the latest commit